episodes/2-identify-the-problem.Rmd

```diff
@@ -133,29 +122,25 @@ rodents <- rodents %>% filter(taxa == "Rodent")
+4.  Reading through the documentation's Description, Usage, Arguments, Details and Examples entries for greater insight into our error.
 
+And, when all else fails, preparing our code into a reproducible example for expert help. Note, there are fewer options available with semantic errors as when an error message prevents your code from running. You may find yourself having to isolate or ask for help more often than with easily-solvable syntax errors.
 
-::::::::::::::::::::: callout
+::: callout
+Generally, the more your code deviates from just using base R functions, or the more you use specific packages, both the quality of documentation and online help available from search engines and Googling gets worse and worse. While base R errors will often be solvable in a couple of minutes from a quick `?help` check or a long online discussion and solutions on a website like Stack Overflow, errors arising from little-used packages applied in bespoke analyses might merit isolating your specific problem to a reproducible example for online help, or even getting in touch with the developers! Such community input and questions are often the way packages and documentation improves over time.
+:::
 
-Generally, the more your code deviates from just using base R functions, or the more you use specific packages, both the quality of documentation and online help available from search engines and Googling gets worse and worse. While base R errors will often be solvable in a couple of minutes from a quick `?help` check or a long online discussion and solutions on a website like Stack Overflow, errors arising from  little-used packages applied in bespoke analyses might merit isolating your specific problem to a reproducible example for online help, or even getting in touch with the developers! Such community input and questions are often the way packages and documentation improves over time. 
+## 3.3 How can I find where my code is failing?
 
-:::::::::::::::::::::
-
-## 3.3 How can I find where my code is failing? 
-
-Isolating your problem may not be as simple as assessing the output from a single function call on the line of code which produces your error. Often, it may be difficult to determine which line(s) in your code are producing the error. 
-
-Consider the example below, where we now are attempting to see how which species of kangaroo rodents appear in different plot types over the years. 
+Isolating your problem may not be as simple as assessing the output from a single function call on the line of code which produces your error. Often, it may be difficult to determine which line(s) in your code are producing the error.
 
+Consider the example below, where we now are attempting to see how which species of kangaroo rodents appear in different plot types over the years. To do so, we'll filter our dataset to just include the genus *Dipodomys*, and then plot a histogram of which how many observations are seen in each plot type over an x axis of years.
```

Comment:
kangaroo rodents

Reply:
I'm not understanding this comment, apologies 😪

Reply:
Oh I just noticed it said "kangaroo rodents" when I think you probably meant either "kangaroo rats" or "rodents"